pyramid/directory.go

```diff
@@ -78,9 +74,20 @@ func (d *directory) renameFile(src, dst string) error {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
-	if err := os.MkdirAll(filepath.Dir(dst), os.ModePerm); err != nil {
-		return fmt.Errorf("creating dir: %w", err)
+	if err := d.ensureParentDir(dst); err != nil {
+		return err
 	}
-
 	return os.Rename(src, dst)
 }
+
+func (d *directory) ensureParentDir(path string) error {
+	parentDir := filepath.Dir(path)
+	if parentDir == d.ceilingDir {
```

Comment:
Why this check? MkdirAll returns nil if path exists.

Reply:
Thought this check was required in order not to create the root folder, the following comment:
// ceilingDir is the root directory of the FS - shouldn't never be deleted

Reply:
MkdirAll is a no-op for existing dirs. IIUC you're saying that you want to expose a possible bug of deleting the ceilingDir by making sure we're not ever creating it? ok then.

Reply:
exactly